diff -Naurp electron-2.0.17/chromium/third_party/webrtc/rtc_base/physicalsocketserver.cc electron-2.0.17.new/chromium/third_party/webrtc/rtc_base/physicalsocketserver.cc
--- electron-2.0.17/chromium/third_party/webrtc/rtc_base/physicalsocketserver.cc	2017-09-22 01:07:33.000000000 +0300
+++ electron-2.0.17.new/chromium/third_party/webrtc/rtc_base/physicalsocketserver.cc	2019-10-12 19:14:02.000000000 +0300
@@ -53,6 +53,10 @@
 #include "webrtc/rtc_base/timeutils.h"
 #include "webrtc/rtc_base/win32socketinit.h"
 
+#if defined(WEBRTC_LINUX)
+#include <linux/sockios.h>
+#endif
+
 #if defined(WEBRTC_POSIX)
 #include <netinet/tcp.h>  // for TCP_NODELAY
 #define IP_MTU 14 // Until this is integrated from linux/in.h to netinet/in.h
