diff --git a/cmake/modules/FindOCIO.cmake b/cmake/modules/FindOCIO.cmake
index 1238220642032965c691a955a01391f40a878245..ea4ef86fdc873e371fb21d4807e655e6171b0927 100644
--- a/cmake/modules/FindOCIO.cmake
+++ b/cmake/modules/FindOCIO.cmake
@@ -27,6 +27,7 @@ if(OCIO_LIBRARY_PATH)
 endif()
 
 find_path(OCIO_INCLUDE_DIR OpenColorIO.h
+        PATHS
         ${OCIO_INCLUDE_PATH}
         ${OCIO_PATH}/include/
         /usr/include
diff --git a/libs/ui/canvas/kis_image_pyramid.cpp b/libs/ui/canvas/kis_image_pyramid.cpp
index f7e46d0b754158f1d2dcd68ff7b8d9424fc42463..ff4ceb02c9aa2146cde62bb62f078e90890058c5 100644
--- a/libs/ui/canvas/kis_image_pyramid.cpp
+++ b/libs/ui/canvas/kis_image_pyramid.cpp
@@ -37,8 +37,8 @@
 
 #include <config-ocio.h>
 #ifdef HAVE_OCIO
-#include <OpenColorIO/OpenColorIO.h>
-#include <OpenColorIO/OpenColorTransforms.h>
+#include <OpenColorIO.h>
+#include <OpenColorTransforms.h>
 #endif
 
 #define ORIGINAL_INDEX           0
diff --git a/plugins/dockers/lut/lutdocker.cpp b/plugins/dockers/lut/lutdocker.cpp
index e00c5a139935e5b7533ae658e9233cf8088e0045..33ce3b652dfc0660caab464a5b7ee194bee3cfa4 100644
--- a/plugins/dockers/lut/lutdocker.cpp
+++ b/plugins/dockers/lut/lutdocker.cpp
@@ -37,7 +37,7 @@
 #include "lutdocker_dock.h"
 #include <KoDockRegistry.h>
 
-#include <OpenColorIO/OpenColorIO.h>
+#include <OpenColorIO.h>
 namespace OCIO = OCIO_NAMESPACE;
 
 
diff --git a/plugins/dockers/lut/lutdocker_dock.h b/plugins/dockers/lut/lutdocker_dock.h
index b8c6671b288b2df95a37a455826a013e9aed272c..5ec44ed2965cd75694fe4466e37aa19662a6d836 100644
--- a/plugins/dockers/lut/lutdocker_dock.h
+++ b/plugins/dockers/lut/lutdocker_dock.h
@@ -32,7 +32,7 @@
 
 #include "ui_wdglut.h"
 
-#include <OpenColorIO/OpenColorIO.h>
+#include <OpenColorIO.h>
 #include "kis_signal_compressor_with_param.h"
 
 
diff --git a/plugins/dockers/lut/ocio_display_filter.h b/plugins/dockers/lut/ocio_display_filter.h
index 48d896a4c07ceec0e16b4580c97cfd332f606722..9e82002370dcbbf5b5321c2b72cc5feab3e78b12 100644
--- a/plugins/dockers/lut/ocio_display_filter.h
+++ b/plugins/dockers/lut/ocio_display_filter.h
@@ -19,8 +19,8 @@
 #define OCIO_DISPLAY_FILTER_H
 
 #include <kis_display_filter.h>
-#include <OpenColorIO/OpenColorIO.h>
-#include <OpenColorIO/OpenColorTransforms.h>
+#include <OpenColorIO.h>
+#include <OpenColorTransforms.h>
 #include <QVector>
 #include "kis_exposure_gamma_correction_interface.h"
 
