--- a/kernel/nv-linux.A	2013-08-23 01:12:13.503534037 +0200
+++ b/kernel/nv-linux.h	2013-08-23 01:12:30.633467230 +0200
@@ -957,7 +957,7 @@
 #endif
 
 #if !defined(NV_VMWARE)
-#define NV_NUM_PHYSPAGES                num_physpages
+#define NV_NUM_PHYSPAGES                get_num_physpages()
 #define NV_GET_CURRENT_PROCESS()        current->tgid
 #define NV_IN_ATOMIC()                  in_atomic()
 #define NV_LOCAL_BH_DISABLE()           local_bh_disable()
